scripts/azureml-assets/azureml/assets/config.py

```diff
@@ -260,12 +260,84 @@ class ModelTaskName(Enum):
     TEXT_CLASSIFICATION = 'text-classification'
 
 
-class ModelDownloadType(Enum):
-    """Enum for Download Method for Remote Model."""
+class PathType(Enum):
+    """Enum for path types supported for model publishing."""
 
-    FTP = 'ftp'
-    GIT = 'git'
-    HTTP = 'http'
+    LOCAL = "local"  # Path to model files present locally.
+    GIT = "git"      # Model hosted on a public GIT repo and can be cloned by GIT LFS.
+    FTP = "ftp"      # <UNSUPPORTED> Model files hosted on a FTP endpoint.
+    HTTP = "http"    # <UNSUPPORTED> Model files hosted on a HTTP endpoint.
+    AZURE = "azure"  # Model files hosted on an AZURE blobstore with public read access.
+
+
+class AssetPath:
+    """Asset path."""
+
+    def __init__(self, asset_type: str, uri: str):
+        """Initialize asset path.
+
+        Args:
+            asset_type (str): path type. Valid values are [local, git, ftp, http, azure].
+            uri (str): a valid URI to local or remote resouce.
+        """
+        self._uri = uri
+        self._type = asset_type
+
+    @property
+    def uri(self) -> str:
+        """Asset URI."""
+        return self._uri
+
+    @property
+    def type(self) -> str:
+        """Asset type."""
+        return self._type
+
+
+class LocalAssetPath(AssetPath):
+    """Local asset path."""
+
+    def __init__(self, uri: str):
+        """Create a Local path of asset.
+
+        Args:
+            uri (str): Path to local model relative to asset.yaml.
+        """
+        super().__init__(PathType.LOCAL, uri=uri)
+
+
+class AzureBlobstoreAssetPath(AssetPath):
+    """Azure Blobstore asset path."""
+
+    BLOBSTORE_URI = "https://{}.blob.core.windows.net/{}/{}"

```

Comment:
> BLOBSTORE_URI = "https://{}.blob.core.windows.net/{}/{}" [](http://example.com/codeflow?start=3&length=57)

Could you please create an issue to support ABFS and WASB and add a TODO here? 

[https://learn.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-abfs-driver](https://learn.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-abfs-driver)

Reply:
Done.